### PR TITLE
Create program.c

### PR DESCRIPTION
**Pull request by kaabilcoder**
The N-Queens Problem is a classic problem where the objective is to place N queens on an N x N chessboard such that no two queens can attack each other. A queen can attack any other piece in the same row, column, or diagonal, making it challenging to place all N queens without conflict.

### CODE REVIEW

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please carefully read our contribution guidelines for docs and we need only `.md` files for our docs so, you can delete your `docs/backtracking algorithms/program.c` file

**Comment by kaabilcoder** [comment]
got it

File: docs/backtracking algorithms/N-QueensProblem.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# N-Queens Problem
```
Comment (ajay-dhangar) [inline docs/backtracking algorithms/N-QueensProblem.md]: Please follow our format and syntax

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Update your code based on our suggestion

File: docs/backtracking algorithms/N-QueensProblem.md (hunk 1 above)
Comment (kaabilcoder) [reply docs/backtracking algorithms/N-QueensProblem.md]: I have updated the code based on your suggestions. please check it out

File: docs/backtracking algorithms/N-QueensProblem.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
+---
+id: N-Queens Problem
```
Comment (ajay-dhangar) [inline docs/backtracking algorithms/N-QueensProblem.md]: replace `id: N-Queens Problem` with `id: n-queens-problem`

File: docs/backtracking algorithms/N-QueensProblem.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+1. **isSafe(row, col)**: Checks if placing a queen at `(row, col)` is safe.
+2. **solveNQueens(row)**: Recursively places queens row by row and uses backtracking when necessary.
+3. **printSolution()**: Displays a valid chessboard configuration.
+
+## Time Complexity
+
+The time complexity for solving the N-Queens problem using backtracking is `O(N!)`, since we try placing a queen in every column of each row and backtrack if needed.
+
+## Program: 
+```
```
Comment (ajay-dhangar) [inline docs/backtracking algorithms/N-QueensProblem.md]: replace line 43 ```` ``` ```` with ```` ```c ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/backtracking algorithms/N-QueensProblem.md (hunk 3 above)
Comment (kaabilcoder) [reply docs/backtracking algorithms/N-QueensProblem.md]: Thank you for your suggestions! i have updated the code based on your suggestions, please check it out

Comment (kaabilcoder) [reply docs/backtracking algorithms/N-QueensProblem.md]: Hello sir! by mistake, i have clicked the unassigned which makes me removed from my assignment. Please reassign this issue to me and review my code

**Comment by kaabilcoder** [comment]
Thank you sir for your suggestion! I have updated the changes which you
have mentioned so please review the check once and also by mistakely i have
been removed from the assignee by mistakely clicked the unassign button. So
i request to reassign this issue to me again and review my code.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**